### PR TITLE
Comment all of the things

### PR DESCRIPTION
**Pull request by christian-marie**

### CODE REVIEW

File: lib/Network/OAuth2/Server.hs [hunk 1]
```diff
@@ -27,8 +43,6 @@ import           Network.OAuth2.Server.Statistics
 import           Paths_oauth2_server                 as P
 
 
--- * Server
```
Comment (thsutton) [inline lib/Network/OAuth2/Server.hs]: y u hate headings? :-(

Comment (christian-marie) [inline lib/Network/OAuth2/Server.hs]: It just seemed so lonely, and didn't seem to match the things below it.

File: lib/Network/OAuth2/Server/API.hs [hunk 2]
```diff
@@ -7,6 +23,14 @@
 {-# LANGUAGE RecordWildCards     #-}
 {-# LANGUAGE TypeOperators       #-}
 
+-- | Anchor specific OAuth2 implementation.
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: It isn't all that Anchor-specific excepting "shibboleth" and that's really just "auth by a front-end proxy"; `mod_ldap` can do/does the same thing we expect.

File: lib/Network/OAuth2/Server/API.hs [hunk 3]
```diff
@@ -7,6 +23,14 @@
...
+--
+-- This implementation assumes the use of Shibboleth, which doesn't actually
+-- mean anything all that specific. This just means that we expect a particular
+-- header that says who the user is.
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: "who the use is and what permissions they have to delegate."

File: lib/Network/OAuth2/Server/API.hs [hunk 4]
```diff
@@ -57,26 +84,41 @@ import           Network.OAuth2.Server.UI
 logName :: String
 logName = "Anchor.Tokens.Server.API"
 
+
+-- TODO: Move this into some servant common package
+
+-- | http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9.2
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: Something like we have for NoCache below would be good.

File: lib/Network/OAuth2/Server/API.hs [hunk 5]
```diff
@@ -57,26 +84,41 @@ import           Network.OAuth2.Server.UI
...
+--
+-- The purpose of the no-store directive is to prevent the inadvertent release
+-- or retention of sensitive information (for example, on backup tapes).
 data NoStore = NoStore
 instance ToByteString NoStore where
     builder _ = "no-store"
 
+-- | Same as Cache-Control: no-cache, we use Pragma for compatibilty.
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: Here's the RFC link: http://tools.ietf.org/html/rfc2616#section-14.32

Comment (tvh) [inline lib/Network/OAuth2/Server/API.hs]: https://tools.ietf.org/html/rfc6749#section-5.1

The authorization server MUST include the HTTP "Cache-Control"
   response header field [RFC2616] with a value of "no-store" in any
   response containing tokens, credentials, or other sensitive
   information, as well as the "Pragma" response header field [RFC2616]
   with a value of "no-cache".

File: lib/Network/OAuth2/Server/API.hs [hunk 6]
```diff
@@ -87,12 +129,17 @@ tokenEndpoint conf auth (Right req) = do
         Right response -> do
             return $ addHeader NoStore $ addHeader NoCache $ response
 
+-- Check that the request is valid, if it is, provide an 'AccessResponse',
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: |

File: lib/Network/OAuth2/Server/API.hs [hunk 7]
```diff
@@ -127,12 +174,17 @@ processTokenRequest ref t (Just client_auth) req = do
...
     return $ grantResponse t access_details (Just $ tokenDetailsToken refresh_details)
 
+-- | Headers for Shibboleth, this tells us who the user is and what they're
+-- allowed to do.
 type OAuthUserHeader = "Identity-OAuthUser"
 type OAuthUserScopeHeader = "Identity-OAuthUserScopes"
 
+-- | The user may want to delete a token, or create a new one with a given
+-- scope.
 data TokenRequest = DeleteRequest
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: Should this move to `Types`?

File: lib/Network/OAuth2/Server/API.hs [hunk 8]
```diff
@@ -253,6 +306,11 @@ handleShib
 handleShib f (Just u) (Just s) = f u s
 handleShib _ _        _        = error "Expected Shibbloleth headers"
 
+-- | Authorize all of the things. This serves the page that allows the user to
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: "Receive requests from clients and display the authorization user interface"

File: lib/Network/OAuth2/Server/API.hs [hunk 9]
```diff
@@ -294,6 +352,7 @@ authorizeEndpoint pool user_id permissions rt c_id' redirect sc' st = do
     request_code <- liftIO $ storeCreateCode pool user_id clientClientId redirect' sc st
     return $ renderAuthorizePage request_code
 
+-- | Handle the response from the page served in 'authorizeEndpoint'
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/API.hs]: s/the response/the approval or rejection/

?

File: lib/Network/OAuth2/Server/API.hs (hunk 7 above)
Comment (christian-marie) [inline lib/Network/OAuth2/Server/API.hs]: Possibly. This is just a doc fix though, to help merging.

File: lib/Network/OAuth2/Server/Statistics.hs [hunk 10]
```diff
@@ -24,7 +33,7 @@ import           Network.OAuth2.Server.Types
 statsLogName :: String
 statsLogName = "Tokens.Server.Statistics"
 
-
+-- | Counters for EKG monitoring
```
Comment (thsutton) [inline lib/Network/OAuth2/Server/Statistics.hs]: "The counters are not added to the EKG server state but will be collected on demand by gatherStats."

File: lib/Network/OAuth2/Server/API.hs (hunk 5 above)
Comment (christian-marie) [inline lib/Network/OAuth2/Server/API.hs]: Dropped in both of these links.

File: lib/Network/OAuth2/Server/Statistics.hs (hunk 10 above)
Comment (christian-marie) [inline lib/Network/OAuth2/Server/Statistics.hs]: Done

**Pull request merged**